pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -277,7 +278,11 @@ func (s *Service) CheckHealth(ctx context.Context, req *backend.CheckHealthReque
 	if err != nil || metricsRes.StatusCode != 200 {
 		status = backend.HealthStatusError
 		if err != nil {
-			return nil, err
+			if _, ok := err.(types.AzureError); ok {
```

Comment:
the idiomatic way of doing this is with the method `.Is` (e.g. `errors.Is(err, types.AzureError)`. For that, you need to change how you return the error and how you define it (see the other comments).